pkg/util/util.go

```diff
@@ -0,0 +1,37 @@
+package util
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// MegaByte denotes one megabyte and is intended to be used as a limit in `ReadAtMost`.
+const MegaByte = 1 << 20
+
+// ReadAtMost reads from reader and errors out if the specified limit (in bytes) is exceeded.
+func ReadAtMost(reader io.Reader, limit int) ([]byte, error) {
+	readBuffer := bufio.NewReader(reader)
+	writeBuffer := bytes.Buffer{}
+
+	bytesRead := 0
+	for bytesRead < limit+1 {
+		b, err := readBuffer.ReadByte()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		writeBuffer.WriteByte(b)
+		bytesRead++
+	}
+
+	if bytesRead > limit {
+		return nil, errors.Errorf("exceeded maximum allowed size of %d bytes", limit)
```

Comment:
If this is going to be a public API, it should be an error recognizable by callers (error type, I guess).

Reply:
Good point. I think we should move it to `./internal`. There we could define the constants and don't need to care (much) about providing something ~generic.

Reply:
Sure, `internal` sounds good. Still a subpackage in there, I guess, but `internal/util` would be perfectly fine.